ndk/src/media/media_codec.rs

```diff
@@ -269,8 +284,7 @@ impl MediaCodec {
         }
     }
 
-    /// Returns [`None`] if timeout is reached.
-    pub fn dequeue_input_buffer(&self, timeout: Duration) -> Result<Option<InputBuffer>> {
+    pub fn dequeue_input_buffer(&self, timeout: Duration) -> Result<InputBuffer> {
```

Comment:
Shouldn't this then return `MediaCodecResult` if `AMediaCodec_dequeueInputBuffer` has the ability to return `MediaCodecInfo::TryAgainLater`?

Reply:
The thing is, the source code is not clear in this regard, and every sample I've seen online does not handle TryAgainLater for the dequeueInputBuffer call.